docs/design/2022-03-25-non-transactional-statement.md

```diff
@@ -0,0 +1,148 @@
+# Non-transactional statements
+
+- Author(s): [Ziqian Qin](http://github.com/ekexium)
+- Tracking issue: https://github.com/pingcap/tidb/issues/33485
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Motivation or Background](#motivation-or-background)
+* [Detailed Design](#detailed-design)
+* [Test Design](#test-design)
+    * [Functional Tests](#functional-tests)
+    * [Scenario Tests](#scenario-tests)
+    * [Compatibility Tests](#compatibility-tests)
+    * [Benchmark Tests](#benchmark-tests)
+* [Impacts & Risks](#impacts--risks)
+* [Investigation & Alternatives](#investigation--alternatives)
+* [Unresolved Questions](#unresolved-questions)
+
+## Introduction
+
+A new syntax that transforms a given DML into a serial execution of mutually exclusive collectively exhaustive statements. 
+
+For example, `split on a limit 1000 delete from t` can be transformed into `delete from t where a between 1 and 1000` and `delete from t where a between 1001 and 2000`, assuming the min and max of `a` are respectively 1 and 2000.
```

Comment:
1. why does not use the sql `DELETE FROM t limit 1000` and executes it circularly? If so, we don't need get the rows by `SELECT` statment.
2. The new syntax is intended to delete all data in table ? Why not truncate?

Reply:
1. It's not an extensible way for update or `insert into select` that may be supported
2. The delete can have a where clause, so truncate doesn't work here.

Reply:
1.It's not an extensible way for update or insert into select that may be supported
>>> As I know, update ... limit 1000, insert into select ....limit 1000 are supported by mysql, I think TiDB supports too.

Reply:
Simply executing `update ... limit x` for N times will result in the update of the same x rows for N times.

Reply:
sql "split on id limit 1000 update t2 set id2 = id2 + 1", which is the same as " update t2 set id2 = id2 + 1 order by id limit 0, 1000; update t2 set id2 = id2 + 1 order by id limit 1000, 2000;..........." and execute it circularly.
What I means is useing limit for delete or update statement can be more effective because it's not necessary to select all data from table. Well, maybe TiDB executes `update ... limit ` or `delete limit` using select command to get all data too, If so, the realization of `split on .. dml` maybe ok.

Reply:
OK I get it. I think both methods don't have to read all data from the table as long as the condition we add can be pushed down to tikv.
The benefit of your idea is to save the first query. But I doubt `offset x limit y` results in reading all data before x.
And things become subtle when there are concurrent writes.